### DISCUSSION TITLE
Cannot detect scheduled JobRunr tasks in Spring MVC

### DISCUSSION THREAD

**Discussion by JonBetts** [jobrunr/jobrunr#116] [Q&A] [answered]
Hi there,

I didn't want to create an issue for this as it is more likely my lack of understanding than an actual issue itself! I have however created a Stack Overflow question for my issue [here](https://stackoverflow.com/questions/67286507/how-do-i-get-jobrunr-to-detect-my-scheduled-background-job-in-a-spring-controlle). That has been active for about a week with no replies, so thought I'd ask for some help here!

Basically, I want to schedule some JobRunr tasks at certain times (i.e. via cron rules), but I don't know how to get my application to detect that there are scheduled jobs that need running/queuing at certain points. Previously I have used the Spring @Scheduled annotation, and as soon as the application is initialised, it picks up anything with that annotation, and ensures it is run at whatever point specified. E.g. `@Scheduled(cron="0 0 2 * * *")` will run a job at 2am every day. I thought the JobRunr @Job annotation was equivalent to that, but I have a test job that should run every minute, and it isn't ever hit.

I would really appreciate any help with this issue. My code examples are in the above Stack Overflow question, so I don't want to repeat myself too much here! But yeah, my JobRunr background server is definitely active, it just isn't detecting my scheduled tasks at all.

Thanks in advance!

**Comment by gsabarwal-code** [comment, marked as answer]
Hello,

@Job doesnt register the job with Jobrunr. You need to do something like the following:

 BackgroundJob.scheduleRecurrently(
                    "print-hello",
                    SampleJobService::printHello,
                    Cron.minutely()

This block of code can be placed within the main method of your application or within a @PostConstruct so that it gets executed during initialization of the application

**Reply by JonBetts** [reply]
Thank you very much for this. I wasn't aware of the `@PostConstruct` annotation, and that does exactly what I need it to 😄

**Comment by rdehuyss** [comment]
Hi,

I just saw you stackoverflow page a couple of days ago.

The answer of @gsabarwal-code is indeed correct. You can also have a look at the various example projects. I'll also try to add it to the documentation of JobRunr.

Cheers!

**Reply by JonBetts** [reply]
Thanks again for the response on this Ronald. I knew it was me being a bit silly rather than any fault of library! I'm looking forward to properly making use of it now. Thank you for your creation!

**Reply by rdehuyss** [reply]
The documentation has been updated! See https://www.jobrunr.io/en/documentation/background-methods/recurring-jobs/#registering-your-recurring-jobs

Thanks again for bringing this to my attention... cheers!

**Reply by yanan-xu** [reply]
Hi there, thanks for the doc, but will Jobrunr add more annotations to make it convenient for Spring integration? 
like `@RecurringJob` `@ScheduledJob` to mark on service methods?

**Reply by rdehuyss** [reply]
Hi @yanan-xu , the `@RecurringJob` indeed makes sense but the `@ScheduledJob` does not make sense in my eyes. Typically, you want to schedule a job after a certain event - e.g. 1 day after a user subscribed for example. Or how do you see it?

**Reply by rdehuyss** [reply]
@yanan-xu - the following JobRunr release will contain an `@Recurring` annotation allowing you to schedule recurring jobs. Thanks for the idea!

**Reply by yanan-xu** [reply]
@rdehuyss  Great, that sounds good to me. Thanks for you reply and quickly actions.
